protocol-specs/generators/multisignature/index.js

```diff
@@ -0,0 +1,221 @@
+	const optionalKeysBuffer = Buffer.from(optionalKeys.join(''), 'utf8');
+	const assetBuffer = Buffer.concat([
+		intToBuffer(mandatoryKeys.length, 1),
+		mandatoryKeysBuffer,
+		intToBuffer(optionalKeys.length, 1),
+		optionalKeysBuffer,
+		intToBuffer(numberOfSignatures, 1),
+	]);
+
+	return assetBuffer;
+};
+
+const createSignatureObject = (txBuffer, account) => ({
+	// publicKey: account.publicKey,
+	signature: signData(
+		hash(Buffer.concat([hexToBuffer(networkIdentifier), txBuffer])),
+		account.passphrase,
+	),
+});
+
+const findAccountByKey = key =>
+	accounts.filter(account => account.publicKey === key);
+
+const generateValidMultisignatureRegistrationTransaction = () => {
+	// basic transaction
+	const tx = {
+		senderPublicKey: accounts[1].publicKey,
+		timestamp: 77045780,
+		type: 12,
+		asset: {
+			mandatoryKeys: [accounts[3].publicKey, accounts[2].publicKey],
+			optionalKeys: [accounts[4].publicKey, accounts[5].publicKey],
+			numberOfSignatures: 3,
+		},
+	};
+
+	sortKeysDescending(tx.asset.mandatoryKeys);
+	sortKeysDescending(tx.asset.optionalKeys);
+
+	const { signatures, ...registrationTx } = tx;
+	// Start building buffer
+	const transactionTimestamp = Buffer.alloc(4);
+	transactionTimestamp.writeIntBE(tx.timestamp, 0, 4);
+	let txBuffer = Buffer.concat([
+		Buffer.alloc(1, registrationTx.type),
+		transactionTimestamp,
+		hexToBuffer(tx.senderPublicKey),
+		intToBuffer(registrationTx.asset.numberOfSignatures, 1),
+		assetToBytes(registrationTx),
+	]);
+
+	registrationTx.signatures = [];
+	let index = 0;
+	// Mandatory keys sign
+	registrationTx.asset.mandatoryKeys.forEach(aKey => {
+		const account = findAccountByKey(aKey).pop();
+		registrationTx.signatures.push({
+			...createSignatureObject(txBuffer, account),
+			index,
+		});
+		index += 1;
+	});
+	// Optional keys sign
+	registrationTx.asset.optionalKeys.forEach(aKey => {
+		const account = findAccountByKey(aKey).pop();
+		registrationTx.signatures.push({
+			...createSignatureObject(txBuffer, account),
+			index,
+		});
+		index += 1;
+	});
+	// Serialize all signatures
+	registrationTx.signatures.forEach(aSignature => {
+		const signatureBuffer = Buffer.concat([
+			Buffer.alloc(1, aSignature.index),
+			hexToBuffer(aSignature.signature),
+		]);
+		txBuffer = Buffer.concat([txBuffer, signatureBuffer]);
+	});
+	// Sender signs whole transaction
+	const signature = signData(
+		hash(Buffer.concat([hexToBuffer(networkIdentifier), txBuffer])),
+		accounts[0].passphrase,
+	);
+
+	const signedRegistrationTx = { ...registrationTx, signature };
+
+	const id = getId(Buffer.concat([txBuffer, Buffer.from(signature, 'hex')]));
+	signedRegistrationTx.id = id;
+
+	return {
+		input: {
+			account: accounts[1],
+			networkIdentifier,
+			coSigners: [accounts[2], accounts[3], accounts[4], accounts[5]],
+			transaction: registrationTx,
+		},
+		output: signedRegistrationTx,
+	};
+};
+
+generateValidMultisignatureRegistrationTransaction();
```

Comment:
why run here?

Reply:
Left by mistake as it was being use for debugging. Removed.